plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -260,7 +262,11 @@ public Optional<ProjectionApplicationResult<ConnectorTableHandle>> applyProjecti
                 .collect(toImmutableList());
 
         if (handle.getColumns().isPresent()) {
-            Set<JdbcColumnHandle> newColumnSet = ImmutableSet.copyOf(newColumns);
+            Set<JdbcColumnHandle> newColumnSet = newColumns.stream()
+                    // It may happen fresh table handle comes with a columns prepared already.
+                    // In such case it may happen that applyProjection may want to add UPDATE_ROW_ID id, which is created later during the planning.
+                    .filter(column -> !column.getColumnName().equals(UPDATE_ROW_ID))
```

Comment:
What about other synthetic handles like DELETE_ROW_ID, MERGE_ROW_ID? We don't care yet since DefaultJdbcMetadata doesn't provide them (via `getDeleteRowIdColumnHandle`)?

Reply:
There is no constant name for them, because they don't have implementation.